### ISSUE TITLE
wxSpinCtrl incorrect size 

### ISSUE THREAD

**Issue by wphilips** [wxWidgets/Phoenix#1903]
**Operating system**: Fedora 33 
**wxPython version & source**:    
python3-wxpython4-4.0.7-13.fc33.x86_6
wxGTK3-3.0.5.1-2.fc33.x86_64

**Python version & source**:        
python3-3.9.1-1.fc33.x86_64

**Description of the problem**:
When creating a SpinCtrlDouble or SpinCtrl (using wxGlade, but that does not matter) the text part of the widget is much too wide. 
If the width of the widget is set programmatically to a more reasonable value. lots of errors appear:
(wxglade_out.py:73312): Gtk-CRITICAL **: 16:03:44.756: gtk_box_gadget_distribute: assertion 'size >= 0' failed in GtkSpinButton
but the appearance is correct

The attached screenshot "bad.png" shows much too wide text next to "Step" and next to "Store Position"
The other screenshot "good.png" shows a more reasonable width, created by progammatically changing the 
size after widget creation. However, to make this work, it seems an inccorrect width has to be passed, probably 
to accomodate for the + and - buttons. 

The attached script reproduces the bad case. If you uncomment the two SetSize calls, you get the good case.
I tried reproducing this with wxPython4.1.1. I was abale to compile it, but it does not run complainign about a missing
_core.so (which indeed does not exist after building)
Buid commands:
git submodule update --init --recursive
build.py dox build
and then settign and exporting PYTHONPATH and LD_LIBRARY_PATH

```python
# Put code sample here
```

**Comment by wphilips** [comment]
Screenshot of the bad layout
![bad](https://user-images.githubusercontent.com/2874424/104847177-db4a5d80-58de-11eb-9575-69fd653c20e5.png)

Screenshot of the expected layout (approximately)
![good](https://user-images.githubusercontent.com/2874424/104847176-dab1c700-58de-11eb-9343-2e42cc96e43d.png)


Script with which screenshot was produced.
[wxglade_out.py.txt](https://github.com/wxWidgets/Phoenix/files/5826298/wxglade_out.py.txt)

**Comment by swt2c** [comment]
Your reproducer is missing code:

```
$ python3 ~/Downloads/wxglade_out.py.txt 
Traceback (most recent call last):
  File "/home/talbert/Downloads/wxglade_out.py.txt", line 13, in <module>
    from xxx import  _get_size_by_chars
ModuleNotFoundError: No module named 'xxx'
```

I suspect things will work much better in wxPython 4.1.1 as in wxWidgets 3.1.x, there have been changes to SpinCtrl to take into account the min/max values when setting the width of the entry portion of the control.  Unfortunately SetSize() attempts to change the overall size of the widget, not taking into account the entry portion vs the button portion.

**Comment by wphilips** [comment]
[wxglade_out.py.txt](https://github.com/wxWidgets/Phoenix/files/5898057/wxglade_out.py.txt)
I have added the missing code to the reproducer script.

**Comment by swt2c** [comment]
Indeed, things work much better with wxPython 4.1.x.  This is what your reproducer looks like.  As you can see, the SpinCtrlDouble is still clipped though and Gtk still complains about GtkSpinCtrl size issues.
![Screenshot from 2021-01-31 10-07-37](https://user-images.githubusercontent.com/971544/106388477-ad6c1b00-63ac-11eb-9b3b-d36a15b6bf26.png)

If I then remove the wx.FIXED_MINSIZE from your sizer.Add calls, I then get this which looks pretty good to me:
![Screenshot from 2021-01-31 10-08-24](https://user-images.githubusercontent.com/971544/106388555-ff14a580-63ac-11eb-9314-87e17d2baaf0.png)
